hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/ddl/HMSDDLExecutor.java

```diff
@@ -192,18 +194,31 @@ public void addPartitionsToTable(String tableName, List<String> partitionsToAdd)
     LOG.info("Adding partitions " + partitionsToAdd.size() + " to table " + tableName);
     try {
       StorageDescriptor sd = client.getTable(databaseName, tableName).getSd();
-      List<Partition> partitionList = partitionsToAdd.stream().map(partition -> {
+      if (syncConfig.getIntOrDefault(HIVE_BATCH_SYNC_PARTITION_NUM) <= 0) {
+        throw new HoodieHiveSyncException("batch-sync-num for sync hive table must be greater than 0, pls check your parameter");
+      }
+      List<Partition> partitionList = new ArrayList<>();
+      int batchSyncPartitionNum = syncConfig.getIntOrDefault(HIVE_BATCH_SYNC_PARTITION_NUM);
+      for (int idx = 0; idx < partitionsToAdd.size(); idx++) {
```

Comment:
see if you can make use of `org.apache.hudi.common.util.CollectionUtils#batches` or `org.apache.hudi.common.util.CollectionUtils#batchesAsStream` for this scenario

Reply:
this is a more cool style of writing than what I did before, and done. thanks